prefect_email/credentials.py

```diff
@@ -91,12 +91,40 @@ class EmailServerCredentials(Block):
 
     _block_type_name = "Email Server Credentials"
     _logo_url = "https://images.ctfassets.net/gm98wzqotmnx/3PcxFuO9XUqs7wU9MiUBMg/ca740e27815d15528373aced667f58b9/email__1_.png?h=250"  # noqa
-
-    username: Optional[str] = None
-    password: SecretStr = SecretStr("")
-    smtp_server: Union[SMTPServer, str] = SMTPServer.GMAIL
-    smtp_type: Union[SMTPType, str] = SMTPType.SSL
-    smtp_port: Optional[int] = None
+    _documentation_url = "https://prefecthq.github.io/prefect-email/credentials/#prefect_email.credentials.EmailServerCredentials"  # noqa
+
+    username: Optional[str] = Field(
+        default=None,
+        description=(
+            "The username to use for authentication to the server. "
+            "Unnecessary if SMTP login is not required."
+        ),
+    )
+    password: SecretStr = Field(
+        default=SecretStr(""),
```

Comment:
I think we might want a default factory here to be safe since `SecretStr` is mutable.

Reply:
Don't think it's possible to have a default factory:
```
from pydantic import SecretStr

SecretStr()
```

```

TypeError                                 Traceback (most recent call last)
Cell In[1], line 3
      1 from pydantic import SecretStr
----> 3 SecretStr()

File ~/mambaforge/envs/email/lib/python3.10/site-packages/pydantic/types.py:888, in pydantic.types.SecretStr.__init__()

TypeError: __init__() takes exactly 2 positional arguments (1 given)
```

Reply:
I think you could make one with something along the lines of `partial(SecretStr, "")`

Reply:
Clever!
```
from functools import partial
from pydantic import SecretStr, Field, BaseModel

class Test(BaseModel):

    password: SecretStr = Field(default_factory=partial(SecretStr, ""))

Test()
```